### DISCUSSION TITLE
feature: automatically generate collisions from tiled data parsed by the tilemap extension

### DISCUSSION THREAD

**Discussion by blurymind** [4ian/GDevelop#2366] [Enhancement ideas (and their technical discussions)]
BEFORE opening a new feature request, please make sure that you:

-   Discussed it on the discord or the forum,
-   There is not already a suggestion about it in the issues or in the roadmap: https://trello.com/b/qf0lM7k8/gdevelop-roadmap
-   Consider commenting on the roadmap if something is important for you

AFTER opening the feature request, the issue will be closed by a maintainer (@4ian or someone else) and a card will be added in the roadmap if it's relevant and does not exist yet :)

## Description

Currently gdevelop can parse tiled files  and render tilemaps from them.
The user however still needs to manually create collision objects within gdevelop.

## Solution suggested

**Describe the solution**
Tiled lets you create shape objects and collisions on tiles.
The tilemap extension can use this data to create collision shape objects - potentially the user needs to be able to
- enable/disable this feature - for shape objects and/or tile shapes (more expensive if each tile has a collision - so not sure if we want to allow that)
- only create collisions for shape objects that are in a specific group, maybe even detect the tiled data group a collision belongs to when colliding with it
- potentially the optional ability to preview collisions generated by the tilemap extension's parser - in the IDE

## Alternatives considered

trello board card here: https://trello.com/c/3B6otFoa/481-add-ability-to-automatically-generate-collision-shapes-to-the-tilemap-extension

**Comment by tristanbob** [comment]
Has anyone made a bounty for this? If so, I would like to contribute to it.

**Reply by tristanbob** [reply]
I think this has to be a Github issue in order for BountySource to see it.  Is that right?

**Reply by 4ian** [reply]
Yep we would need to open a GitHub issue. I've done it here: https://github.com/4ian/GDevelop/issues/2775
